### PR TITLE
Refactor the build system

### PR DESCRIPTION
**Pull request by Ecco**
- Use variants (e.g. `src += foo/bar.cpp:+variant`)
- macOS and iOS can now build any binary (e.g. `make test.app`)
- Streamline the root Makefile
- Share multi-arch Make code
- Get rid of DEFAULT target in favor of first HANDY_TARGET
- Add extra (and cheap) targets to the CI

### CODE REVIEW

**Comment by Ecco** [comment]
Make sure to compare the results of all build targets before merging (including bench targets).

File: build/targets.simulator.ios.mak [hunk 1]
```diff
@@ -0,0 +1,10 @@
+ifndef ARCH
+HANDY_TARGETS_EXTENSIONS += apk
```
Comment (EmilieNumworks) [inline build/targets.simulator.ios.mak:2]: ipa?

Comment (Ecco) [reply build/targets.simulator.ios.mak:2]: Well, not even that. Those are in `build/platform.*.mak` now 😄

**Comment by Ecco** [comment]
- Comparing all_official seems ok (minor differences, at least due to the commit being embedded). By the way, the all_official target is broken. PR incoming.
- Comparing binpacks seems ok too (again, super small differences, a few bytes).

**Comment by RedGl0w** [comment]
@Ecco it seems that moving the `SFLAGS` to default.mak is definitely not a good idea... Now, even when you just compile the linux simulator, the `EPSILON_GETOPT` is true, and so it causes https://github.com/numworks/epsilon/issues/1493

**Pull request merged**